src/main/java/ir/sahab/kafkaconsumer/OffsetTracker.java

```diff
@@ -56,17 +64,29 @@
      *                                 partition, next calls to {@link #ack(int, long)} on that
      *                                 partition fails and returns false.
      */
-    public OffsetTracker(int pageSize, int maxOpenPagesPerPartition) {
+    public OffsetTracker(int pageSize, int maxOpenPagesPerPartition, MetricRegistry metricRegistry) {
         this.pageSize = pageSize;
         this.maxOpenPagesPerPartition = maxOpenPagesPerPartition;
+        this.metricRegistry = metricRegistry;
         this.partitionTrackers = new HashMap<>();
     }
 
+
+    /**
```

Comment:
Remove this constructor. Instead pass new MetricRegistry() from test code.

Reply:
Done